core/src/main/java/org/apache/iceberg/DataFiles.java

```diff
@@ -301,12 +299,10 @@ public Builder withSplitOffsets(List<Long> offsets) {
       return this;
     }
 
+    /** @deprecated since 1.5.0, will be removed in 1.6.0; must not be set for data files. */
+    @Deprecated
     public Builder withEqualityFieldIds(List<Integer> equalityIds) {
-      if (equalityIds != null) {
-        this.equalityFieldIds = ImmutableList.copyOf(equalityIds);
-      }
-
-      return this;
+      throw new UnsupportedOperationException("Equality field IDs must not be set for data files");
```

Comment:
[optional] should we add an UT to assert the exception is thrown when this API is used ?

Reply:
Yeah, let me add it.

Reply:
Actually, IntelliJ starts to warn that the deprecated method is being used. We also don't have a good suite for the data file builder. Adding a new suite is probably not worth the effort. I'd skip it.